web/src/app/forms/FormField.js

```diff
@@ -186,13 +186,24 @@ export class FormField extends React.PureComponent {
           error={checkbox ? undefined : Boolean(props.error)}
           label={this.props.formLabel ? null : props.label}
         />
-        {!noError && (props.error || props.hint) && (
-          <FormHelperText>
-            {(props.error &&
-              props.error.message.replace(/^./, (str) => str.toUpperCase())) ||
-              props.hint}
-          </FormHelperText>
-        )}
+        {!noError &&
+          (props.error || props.hint) &&
+          typeof props.error.message === 'string' && (
+            <FormHelperText>
+              {(props.error &&
+                props.error.message.replace(/^./, (str) =>
+                  str.toUpperCase(),
+                )) ||
+                props.hint}
+            </FormHelperText>
+          )}
+        {!noError &&
+          (props.error || props.hint) &&
+          typeof props.error.message !== 'string' && (
+            <FormHelperText>
+              {(props.error && props.error.message) || props.hint}
+            </FormHelperText>
+          )}
```

Comment:
Rather than adding multiple ways of rendering errors, we should make this new one follow existing conventions